os/fs/vfs/fs_select.c

```diff
@@ -265,6 +265,10 @@ int select(int nfds, FAR fd_set *readfds, FAR fd_set *writefds, FAR fd_set *exce
 	int ndx = _init_desc_list(nfds, readfds, writefds, exceptfds, pollset);
 
 	DEBUGASSERT(ndx == npfds);
+	if (ndx != npfds) {
+		set_errno(EINVAL);
```

Comment:
we return EINVAL if any of parameter is invalid but both of ndx & npfds are not parameter of this function.
    This happened when fdset is invalid, then EBADF is correct value I think.